uwblueprint/paramedics-react
# Grey out button for invalid team member fields

[Notion ticket](https://www.notion.so/uwblueprintexecs/Resource-management-new-team-member-Done-button-is-not-greyed-out-089cc401605a4125910cfc313d469c32)

[Figma link](https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-S20-F20?node-id=1972%3A35883)

## Changes
- Added checks for valid form fields when adding a new team member
- Field checks for member name and valid email
- Added a way to get validator results from the component FormFields
- Default the role selection to Supervisor

## Screenshots
<img width="1204" alt="Screen Shot 2020-11-19 at 12 12 03 PM" src="https://user-images.githubusercontent.com/59406513/99718894-713d4580-2a60-11eb-9234-7027c69e51b6.png">
<img width="1210" alt="Screen Shot 2020-11-19 at 12 12 24 PM" src="https://user-images.githubusercontent.com/59406513/99718921-7dc19e00-2a60-11eb-84b9-2e67962c7938.png">

## Testing
- Open to the team member add flow
- Try the following combination to see if valid checks work:
> Empty name field & valid email field = INVALID
> filled name field & invalid email field (invalid email or email in use) = INVALID
> filled name field & valid email field = VALID

VALID = coloured button, INVALID = greyed out button

## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
